jerry-core/jerry.c

```diff
@@ -2023,42 +2019,19 @@ jerry_parse_and_save_snapshot (const jerry_api_char_t *source_p, /**< script sou
   jsp_status_t parse_status;
   ecma_compiled_code_t *bytecode_data_p;
 
-  snapshot_buffer_write_offset = 0;
-  snapshot_last_compiled_code_offset = 0;
+  snapshot_buffer_write_offset = JERRY_ALIGNUP (sizeof (jerry_snapshot_header_t),
+                                                MEM_ALIGNMENT);
```

Comment:
As we construct the snapshot buffer, we seem to use alignments based on `MEM_ALIGNMENT` and `MEM_ALIGNMENT_LOG` (scattered all around the code, not just here). Do we realize that changing those defines invalidates all our snapshots? I mean, there is a `JERRY_SNAPSHOT_VERSION` macro which is incremented every time there is a breaking change to the snapshot format (just like this patch). Memory alignment seems only to be a memory configuration parameter, I'm not sure it should have any effect on the snapshot format.